src/Sulu/Component/DocumentManager/DocumentManager.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\DocumentManager;
+
+use Sulu\Component\DocumentManager\Collection\QueryResultCollection;
+use Sulu\Component\DocumentManager\Events;
+use Sulu\Component\DocumentManager\Event;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class DocumentManager
+{
+    /**
+     * @var EventDispatcher
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var NodeManager
+     */
+    private $nodeManager;
+
+    public function __construct(EventDispatcherInterface $eventDispatcher, NodeManager $nodeManager)
+    {
+        $this->eventDispatcher = $eventDispatcher;
+        $this->nodeManager = $nodeManager;
+    }
+
+    /**
+     * Find a document by path or UUID in the given
+     * locale, optionally enforcing the given type.
+     *
+     * @param string $id Path or UUID
+     * @param string $locale Locale
+     * @param string $aliasOrClass Document alias or class
+     */
+    public function find($id, $locale, $aliasOrClass = null)
+    {
+        $event = new Event\FindEvent($id, $locale, $aliasOrClass = null);
+        $this->eventDispatcher->dispatch(Events::FIND, $event);
+
+        return $event->getDocument();
+    }
+
+    /**
+     * Create a new document instance for the given alias
+     *
+     * @param string
+     * @throws MetadataNotFoundException
+     * @return object
+     */
+    public function create($alias)
+    {
+        $event = new Event\CreateEvent($alias);
+        $this->eventDispatcher->dispatch(Events::CREATE, $event);
+
+        return $event->getDocument();
+    }
+
+    /**
+     * Persist a document to a PHPCR node
+     *
+     * @param object $document
+     * @param string $locale
+     */
+    public function persist($document, $locale)
```

Comment:
This actually "flushes" the document to the PHPCR node - it is more like a "snapshot". Changes made to the Document afterwards will NOT be taken into account when `flush` is called.

Maybe rename to something more appropriate and less confusing.